ckanext/dcat/harvesters/_json.py

```diff
@@ -382,3 +381,9 @@ def copy_across_resource_ids(existing_dataset, harvested_dataset):
                     matching_existing_resource)
         if not existing_resources_still_to_match:
             break
+
+    config = config if config is not None else {}
+    if config.get('keep_existing_resources', False):
```

Comment:
The statement here is checking if `keep_existing_resources` exists and not evaluating it as a boolean. If a user sets `keep_existing_resources` to the string`"False"` existing resources will be kept.

You should add `keep_existing_resources` to config_processors to validate that a boolean value is set